app/resonant-laboratory/general_purpose/binUtils.js

```diff
@@ -19,7 +19,11 @@ function coerceValue (value, coerceToType) {
   } else if (coerceToType === 'number') {
     value = parseFloat(value);
   } else if (coerceToType === 'string') {
-    value = String(value);
+    if (value && value.str) {
+      value = value.str;
```

Comment:
No, this solution is perfectly good. I was just curious where that `str` field comes from.